.github/workflows/pages.yml

```diff
@@ -0,0 +1,92 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: GitHub pages deployment
+
+on:
+  push:
+    branches:
+      - master
+
+
+concurrency:
+  group: 'docs preview'
+  cancel-in-progress: true
```

Comment:
This makes the build dependent on each other. E.g., if you have commit 1,2,3 at the same time, build 1 has to be finished before starting build 2. Can we maybe just do this as a scheduled job to save resources?

Reply:
> if you have commit 1,2,3 at the same time, build 1 has to be finished before starting build 2

1 and 2 will be canceled by 3.